src/parsec_client/operations/get_make_cred_params.md

```diff
@@ -0,0 +1,60 @@
+# GetMakeCredParams
+
+Get the parameters for performing a `TPM2_MakeCredential` operation. Opcode: 31 (`0x001F`)
+
+**This is a TPM-specific operation.**
```

Comment:
If this has to be a TPM-specific operation, then maybe our opcode naming needs a convention for this, maybe even  making it an opcode name prefix like `TpmGetMakeGredParam`, because this would be the first case where the opcode is entirely specific to the provider. But I wonder whether there is a way to avoid making it explicitly TPM-specific. It feels slightly lopsided to me that `AttestKey` is a generic operation that can accept a TPM-specific flavour of input, whereas `GetMakeCredParam` is an entirely TPM-specific opcode, rather than a generic opcode that can return a TPM-specific flavour of output. Either way, we know that the ultimate client of these APIs will need to have TPM intelligence, and that's okay. But is there a way to make the design more consistent in terms of how that intelligence is exercised? At the moment it looks a bit half-and-half. For `AttestKey` you need to make sure you pass in TPM-oriented parameters, but for `GetMakeCredParams` the whole opcode is TPM-specific.

I'm inclined to suggest that we do one of two things, we either make _both_ of these operations TPM-specific (`TpmAttestKey` and `TpmGetMakeCredParams`), and just dismiss any idea that attestation can ever be provider-neutral. Or, alternatively, remodel `GetMakeCredParams` as something like `PrepareKeyAttestation`, and have a TPM-specific interpretation of the results, which would be precisely analogous to how `AttestKey` has a TPM-specific interpretation of its parameters.

Reply:
Do you have any preference between the two? I'll have to have a think about what either of them mean in terms of developer experience

Reply:
I think both options have an almost identical "footprint" for a client - they'd still need to use mechanism-specific code in both cases, just that in one of them (where both operations are TPM-specific) the client function call would be TPM-specific as well, not just the parameters that get passed to it.

For the service, the downside of generic operations is that providers would need to reject mechanisms they don't support, but would otherwise be cleaner to implement. I'm leaning towards the generic approach since it also suits Parsec goals better.

Reply:
Ok, I've made the 2nd operation generic as well, and wrote up the parameters/output of that in another file, much like for `AttestKey`.